recipes/crashpad/all/conanfile.py

```diff
@@ -0,0 +1,275 @@
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from contextlib import contextmanager
+import os
+import textwrap
+
+required_conan_version = ">=1.33.0"
+
+
+class CrashpadConan(ConanFile):
+    name = "crashpad"
+    description = "Crashpad is a crash-reporting system."
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("conan", "crashpad", "crash", "error", "stacktrace", "collecting", "reporting")
+    license = "Apache-2.0"
+    homepage = "https://chromium.googlesource.com/crashpad/crashpad/+/master/README.md"
+    provides = "crashpad", "mini_chromium"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "fPIC": [True, False],
+        "http_transport": ["libcurl", "socket", None],
+        "with_tls": ["openssl", False],
+    }
+    default_options = {
+        "fPIC": True,
+        "http_transport": None,
+        "with_tls": "openssl",
+    }
+    exports_sources = "patches/*"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.os in ("Linux", "FreeBSD"):
+            self.options.http_transport = "libcurl"
+        elif self.settings.os == "Android":
+            self.options.http_transport = "socket"
+
+    def build_requirements(self):
+        self.build_requires("ninja/1.10.2")
+        self.build_requires("gn/cci.20210429")
+        # FIXME: needs python 2.x support on Windows (uses print without parentheses + _winreg module)
+
+    def requirements(self):
+        # FIXME: use mini_chromium conan package instead of embedded package
+        self.requires("zlib/1.2.11")
+        if self.settings.os in ("Linux", "FreeBSD"):
+            self.requires("linux-syscall-support/cci.20200813")
+        if self.options.http_transport != "socket":
+            del self.options.with_tls
+        if self.options.http_transport == "libcurl":
+            self.requires("libcurl/7.75.0")
+        if self.options.get_safe("with_tls") == "openssl":
+            self.requires("openssl/1.1.1k")
+
+    def validate(self):
+        if self.options.http_transport == "libcurl":
+            if not self.options["libcurl"].shared:
+                # FIXME: is this true?
+                self.output.warn("crashpad needs a shared libcurl library")
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Windows is not (yet) supported by this recipe because the build system requires python 2.x which is not (yet) available on CCI.")
```

Comment:
Maybe it is better to just patch the sources? Recipe for `cpython` is being hard.

Reply:
Is this really still the case? `depot_tools` in recent versions should work fine with python 3.8, which AFAIK is the dependency that had the python issues.

Reply:
It works on Python 3.9.5 on Linux (works on my machine).
But an error is thrown when building on Windows.
Maybe it can be patched..